File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> theSet = new HashSet<Integer>();
        ArrayList<Integer> theList = new ArrayList<Integer>();

        for (int i : nums) {
            theSet.add(i);
            theList.add(i);
        }

        return theSet.size() != theList.size();
    }
}

File: Data Structures & Algorithms/majority-element/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        map = {}

        for num in nums:
            if num in map:
                map[num] += 1
            else:
                map[num] = 1

        print(map)

        for num in map.keys():
            if map[num] > len(nums) / 2:
                return num